frontend/src/metabase/writeback/components/ActionViz/Action.tsx

```diff
@@ -1,23 +1,106 @@
-import React from "react";
+import React, { useMemo, useCallback } from "react";
+import { t } from "ttag";
+import { connect } from "react-redux";
 
 import { isImplicitActionButton } from "metabase/writeback/utils";
 
-import type { ActionDashboardCard, Dashboard } from "metabase-types/api";
+import type {
+  ActionDashboardCard,
+  ArbitraryParameterForActionExecution,
+  ActionParametersMapping,
+  Dashboard,
+  ParameterMappedForActionExecution,
+  WritebackQueryAction,
+} from "metabase-types/api";
 import type { VisualizationProps } from "metabase-types/types/Visualization";
+import type { Dispatch } from "metabase-types/store";
+import type { ParameterValueOrArray } from "metabase-types/types/Parameter";
+
+import {
+  getDashcardParamValues,
+  getNotProvidedActionParameters,
+  prepareParameter,
+} from "metabase/modes/components/drill/ActionClickDrill/utils";
+import { executeRowAction } from "metabase/dashboard/actions";
+import { StyledButton } from "./ActionButton.styled";
 
 import DefaultActionButton from "./DefaultActionButton";
 import ImplicitActionButton from "./ImplicitActionButton";
 
 interface ActionProps extends VisualizationProps {
   dashcard: ActionDashboardCard;
   dashboard: Dashboard;
+  dispatch: Dispatch;
+  parameterValues: { [id: string]: ParameterValueOrArray };
 }
 
-function Action({ dashcard, ...props }: ActionProps) {
+function ActionComponent({
+  dashcard,
+  dashboard,
+  dispatch,
+  isSettings,
+  settings,
+  getExtraDataForClick,
+  onVisualizationClick,
+  parameterValues,
+}: ActionProps) {
```

Comment:
Maybe it'd make sense to break it down a bit further. Explicit actions dispatch actions and need to collect params while implicit actions or nav buttons barely need anything like that. Maybe a top-level `Action` component could act as an entry point, figure out the type of action, and then return implicit / explicit / nav action component. And something like `ExplicitAction` can handle all the params and action dispatching stuff.

Just thinking out loud here mostly — don't want that to get in your way of merging this already great PR